Introducing a separate product cancellation flow for Jetpack Products

jboland88 opened the pull request:
#### Changes proposed in this Pull Request

* This PR will introduce a separate cancellation flow for Jetpack products that allows for steps specific to Jetpack
* First, the cancellation flow will show benefits of the Jetpack plan/ product that will be lost if it is cancelled
* Second, the flow will pose an optional survey question to the user as to why they are cancelling 
* The cancellation flow may make a one-time offer to the user to save on their next plan/ product payment

#### Testing instructions

This PR is IN PROGRESS testing instructions will be updated as more is added

**Visual Testing**
- Make sure your Wordpress.com account has at least one site (ideally a test site) with a Jetpack plan or product
- Visit the calypso.live link that contains this branch (linked by the matticbot below)
- Using the site switcher, select your site and navigate to Upgrades > Purchases
- Select the Individual Jetpack product you want to cancel from this page, this will take you to the individual purchase page
- From the Individual purchase page, click the button at the bottom of the page to remove the purchase:
![Screen Shot 2021-07-20 at 10 11 24 AM](https://user-images.githubusercontent.com/18016357/126338989-f701dafd-f47e-4551-9c8d-ff4b076b1b55.png)
- This will present you with the new cancellation flow modal. Each step in the flow is outlined in more detail below.

Jetpack Benefits Step
When the cancellation flow first opens, the first step will show some of the benefits Jetpack provides for the current plan or product. These can vary by plan or product. Outputs for standalone products like Scan or Backup will show expanded outputs with more information about the product. Unit tests on PR #54947 are testing the logic for which outputs show for which plans and products. Try testing multiple sites with different plans or products to see different outputs.

Jetpack Complete Example:
![image](https://user-images.githubusercontent.com/5714212/131520308-4907f624-3846-49df-85be-8a48012ee54c.png)


Backup Realtime Example:
![image](https://user-images.githubusercontent.com/5714212/131520962-5b1052ed-7b96-4c72-b99d-c9dc9f5c86cb.png)


Cancellation Survey:
![image](https://user-images.githubusercontent.com/5714212/131521041-4e46d4c7-970e-45ff-ad4e-a97839edd79b.png)


Dynamic outputs include:
* **Site Backups**: Will show the time of the last successful backup. If a backup is in progress, will show a progress bar

The top of the panel should show the name of the plan and how much time is left on the subscription (unless the subscription is managed by a host, and then it should just indicate the plan/ product is "still active")

jboland88 commented:
Hello - sorry for the review ping on this one, I have converted it to a draft since this is still too early to review or evaluate.

a8ci18n commented on client/components/marketing-survey/cancel-jetpack-form/index.jsx:
```diff
@@ -0,0 +1,258 @@
+		};
+
+		const cancelText = translate( 'Cancel Now' );
+		const cancellingText = translate( 'Cancelling' );
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **3** times:
`translate( 'Why cancelling?' )` *ES Score: 7*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/improve-jetpack-cancellation-flow)

jboland88 committed:
Add component to show general benefits Jetpack provides by plan/ product

jboland88 committed:
Fix typo in the documentation

jboland88 committed:
Introduce JetpackBenefits component and subcomponents that can be used to show key featues of a Jetpack plan or product

jboland88 committed:
Introduce JetpackBenefits component and subcomponents that can be used to show key featues of a Jetpack plan or product

jboland88 committed:
Add step for Jetpack Cancellation flow to show plan/ product information and benefits

jboland88 committed:
Add step for Jetpack Cancellation flow to show plan/ product information and benefits

jboland88 committed:
Improved the plan/ product checks in JetpackBenefits and added unit tests for these checks

jboland88 committed:
Improved the plan/ product checks in JetpackBenefits and added unit tests for these checks

jboland88 committed:
Add a new cancellation form for jetpack products

jboland88 committed:
Improvements to the scan benefit component, added testing for scan benefits component

jboland88 committed:
Improvements to the scan benefit component, added testing for scan benefits component

jboland88 committed:
Revisions to the backup output for Jetpack benefits, refactor to use hooks, added tests for backups output

jboland88 committed:
Revisions to the backup output for Jetpack benefits, refactor to use hooks, added tests for backups output

jboland88 committed:
Refactor scan benefit card to use hooks, cleanup, add translate calls

jboland88 committed:
Refactor scan benefit card to use hooks, cleanup, add translate calls

jboland88 committed:
Merge branch 'add/new-cancellation-flow-for-jetpack' into update/improve-jetpack-cancellation-flow

jboland88 committed:
Merge branch 'add/jetpack-cancellation-flow-benefits-step' into update/improve-jetpack-cancellation-flow

jboland88 committed:
Merge branch 'add/jetpack-benefits-components' into update/improve-jetpack-cancellation-flow

jboland88 committed:
Merge branch 'add/jetpack-general-benefits-component' into update/improve-jetpack-cancellation-flow

a8ci18n commented on client/blocks/jetpack-benefits/index.jsx:
```diff
@@ -0,0 +1,190 @@
+			<JetpackBenefitsCard
+				headline={ translate( 'Search' ) }
+				description={ translate(
+					'Jetpack Search helps your visitors instantly find the right content – right when they need it.'
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **20** times:
`translate( 'Incredibly powerful and customizable, Jetpack Search, helps your visitors instantly find the right content – right when they need it.' )` *ES Score: 8*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/improve-jetpack-cancellation-flow)

a8ci18n commented on client/blocks/jetpack-benefits/index.jsx:
```diff
@@ -0,0 +1,190 @@
+			<JetpackBenefitsCard
+				headline={ translate( 'Activity Log' ) }
+				description={ translate(
+					'The Activity Log shows a list of management events that have occurred on your site.'
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( 'The activity log shows a full list of management events that have occurred on your site.' )` *ES Score: 8*

a8ci18n commented on client/blocks/jetpack-benefits/scan-history.jsx:
```diff
@@ -0,0 +1,156 @@
+			<JetpackBenefitsCard
+				jestMarker="error"
+				headline={ translate( 'Site Scan' ) }
+				description={ translate( 'Jetpack is having trouble scanning your site.' ) }
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **23** times:
`translate( 'Jetpack is having trouble connecting to your site' )` *ES Score: 6*

a8ci18n commented on client/blocks/jetpack-benefits/scan-history.jsx:
```diff
@@ -0,0 +1,156 @@
+				} }
+				stats={ [
+					{
+						title: translate( 'Threats Found' ),
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **3** times:
`translate( 'Malware threats found' )` *ES Score: 8*

a8ci18n commented on client/blocks/jetpack-benefits/site-backups.tsx:
```diff
@@ -0,0 +1,169 @@
+			<JetpackBenefitsStandaloneCard
+				title={ translate( 'Site Backups' ) }
+				summary={ {
+					title: translate( 'Last Backup' ),
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **21** times:
`translate( 'Last backup was on:' )` *ES Score: 7*

IanRamosC committed:
Marketing Survey: Add a survey before a Jetpack product is cancelled

This is one of the components that will be shown in the new/improved Jetpack cancellation flow.

jboland88 committed:
Refactor benefits components to use typescript

jboland88 committed:
Refactor benefits components to use typescript

jboland88 committed:
A few additional adjustments for typescript

jboland88 committed:
A few additional adjustments for typescript

jboland88 committed:
Merge branch 'add/jetpack-benefits-components' into update/improve-jetpack-cancellation-flow

a8ci18n commented on client/blocks/jetpack-benefits/index.tsx:
```diff
@@ -0,0 +1,184 @@
+			<JetpackBenefitsCard
+				headline={ translate( 'Search' ) }
+				description={ translate(
+					'Jetpack Search helps your visitors instantly find the right content – right when they need it.'
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **20** times:
`translate( 'Incredibly powerful and customizable, Jetpack Search, helps your visitors instantly find the right content – right when they need it.' )` *ES Score: 8*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/improve-jetpack-cancellation-flow)

a8ci18n commented on client/blocks/jetpack-benefits/index.tsx:
```diff
@@ -0,0 +1,184 @@
+			<JetpackBenefitsCard
+				headline={ translate( 'Activity Log' ) }
+				description={ translate(
+					'The Activity Log shows a list of management events that have occurred on your site.'
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( 'The activity log shows a full list of management events that have occurred on your site.' )` *ES Score: 8*

a8ci18n commented on client/blocks/jetpack-benefits/scan-history.tsx:
```diff
@@ -0,0 +1,160 @@
+			<JetpackBenefitsCard
+				jestMarker="error"
+				headline={ translate( 'Site Scan' ) }
+				description={ translate( 'Jetpack is having trouble scanning your site.' ) }
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **23** times:
`translate( 'Jetpack is having trouble connecting to your site' )` *ES Score: 6*

a8ci18n commented on client/blocks/jetpack-benefits/scan-history.tsx:
```diff
@@ -0,0 +1,160 @@
+				} }
+				stats={ [
+					{
+						title: translate( 'Threats Found' ),
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **3** times:
`translate( 'Malware threats found' )` *ES Score: 8*

jboland88 committed:
Refactor CancelJetpackForm component to use hooks and TypeScript

jboland88 committed:
Remove superfluous type declaration in steps.ts

a8ci18n commented on client/components/marketing-survey/cancel-jetpack-form/index.tsx:
```diff
@@ -0,0 +1,243 @@
+		};
+
+		const cancelText = translate( 'Cancel Now' );
+		const cancellingText = translate( 'Cancelling' );
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **3** times:
`translate( 'Why cancelling?' )` *ES Score: 7*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/improve-jetpack-cancellation-flow)

jboland88 committed:
Revisions based on initial feedback

jboland88 committed:
Revisions based on initial feedback

jboland88 committed:
Merge branch 'add/jetpack-benefits-components' into update/improve-jetpack-cancellation-flow

jboland88 committed:
Added a feature flag to conditionally show the Jetpack cancellation flow in testing environments for now

a8ci18n commented on client/blocks/jetpack-benefits/index.tsx:
```diff
@@ -0,0 +1,110 @@
+				<JetpackBenefitsCard
+					headline={ translate( 'Search' ) }
+					description={ translate(
+						'Jetpack Search helps your visitors instantly find the right content – right when they need it.'
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **20** times:
`translate( 'Incredibly powerful and customizable, Jetpack Search, helps your visitors instantly find the right content – right when they need it.' )` *ES Score: 8*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/improve-jetpack-cancellation-flow)

a8ci18n commented on client/blocks/jetpack-benefits/scan-history.tsx:
```diff
@@ -0,0 +1,160 @@
+				} }
+				stats={ [
+					{
+						title: translate( 'Malware Threats Found' ),
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **3** times:
`translate( 'Malware threats found' )` *ES Score: 12*

jboland88 committed:
Change feature flags for Jetpack cancellation flow to only show the feature on development

IanRamosC committed:
Jetpack Benefits: Add translations for the general benefits component

IanRamosC committed:
Jetpack Benefits: Move helper function calls to variables

This will make the component calls the functions only once and it will reuse the values of those variables in all the other conditions

IanRamosC committed:
Jetpack Cancellation Flow: Add translations to the benefits step

This refactors the component to use TypeScript and FunctionComponent, as well as adds translated strings instead of the raw text

IanRamosC committed:
Jetpack Cancellation Flow: Add translations to the benefits step

This refactors the component to use TypeScript and FunctionComponent, as well as adds translated strings instead of the raw text

IanRamosC committed:
Jetpack Cancellation Flow: Address latest feedback

IanRamosC committed:
Jetpack Cancellation Flow: Address latest feedback

IanRamosC committed:
Jetpack Cancellation Flow: Improve unit of time variable name

IanRamosC committed:
Jetpack Cancellation Flow: Improve unit of time variable name

jboland88 committed:
Updates based on feedback

jboland88 committed:
Updates based on feedback

jboland88 committed:
Merge branch 'add/jetpack-benefits-components' into update/improve-jetpack-cancellation-flow

jboland88 committed:
Merge branch 'add/jetpack-cancellation-flow-benefits-step' into update/improve-jetpack-cancellation-flow

jboland88 committed:
Merge branch 'add/jetpack-general-benefits-component' into update/improve-jetpack-cancellation-flow

jboland88 committed:
Correctly type the purchase prop and pass the missing productSlug prop down to subcomponents

a8ci18n commented on client/blocks/jetpack-benefits/general-benefits.tsx:
```diff
@@ -0,0 +1,168 @@
+	if ( hasBusinessPlan || hasSecurityRealTimePlan || hasCompletePlan ) {
+		benefits.push(
+			<React.Fragment>
+				{ translate( 'Unlimited {{strong}}high-speed-video hosting{{/strong}}.', {
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **1** times:
`translate( '{{strong}}Unlimited{{/strong}} Video hosting' )` *ES Score: 6*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/improve-jetpack-cancellation-flow)

a8ci18n commented on client/blocks/jetpack-benefits/scan-history.tsx:
```diff
@@ -0,0 +1,160 @@
+	let cardStat: TranslateResult = '';
+	let cardDescription: TranslateResult = '';
+	if ( threats.length > 0 ) {
+		cardStat = translate( '%(threatCount)d threat found', '%(threatCount)d threats found', {
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **20** times:
`translate( 'Database %(threatCount)d threat', 'Database %(threatCount)d threats' )` *ES Score: 8*

a8ci18n commented on client/components/marketing-survey/cancel-jetpack-form/jetpack-benefits-step.tsx:
```diff
@@ -0,0 +1,159 @@
+
+		if ( timeRemaining.months() >= 1 ) {
+			const timeRemainingNumber = timeRemaining.months();
+			const unitOfTime = translate( 'month', 'months', { count: timeRemainingNumber } );
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **23** times:
`translate( 'month', 'months', { context: 'Used in the trial period dropdown. Number is in text field. 0, 2+ will need plural, 1 will need singular.'} )` *ES Score: 16*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/improve-jetpack-cancellation-flow)

a8ci18n commented on client/components/marketing-survey/cancel-jetpack-form/jetpack-benefits-step.tsx:
```diff
@@ -0,0 +1,159 @@
+			return { timeRemainingNumber, unitOfTime };
+		} else if ( timeRemaining.weeks() >= 1 ) {
+			const timeRemainingNumber = timeRemaining.weeks();
+			const unitOfTime = translate( 'week', 'weeks', { count: timeRemainingNumber } );
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **22** times:
`translate( 'week', 'weeks', { context: 'Used in the trial period dropdown. Number is in text field. 0, 2+ will need plural, 1 will need singular.'} )` *ES Score: 17*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/improve-jetpack-cancellation-flow)

a8ci18n commented on client/components/marketing-survey/cancel-jetpack-form/jetpack-benefits-step.tsx:
```diff
@@ -0,0 +1,159 @@
+		}
+
+		const timeRemainingNumber = timeRemaining.days();
+		const unitOfTime = translate( 'day', 'days', { count: timeRemainingNumber } );
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **23** times:
`translate( 'day', 'days', { context: 'Used in the trial period dropdown. Number is in text field. 0, 2+ will need plural, 1 will need singular.'} )` *ES Score: 14*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/improve-jetpack-cancellation-flow)

jboland88 committed:
Add handling for condition where scan is active and healthy, but has not run yet/ there is not a recent scan

jboland88 committed:
Add handling for condition where scan is active and healthy, but has not run yet/ there is not a recent scan

jboland88 committed:
Merge branch 'add/jetpack-benefits-components' into update/improve-jetpack-cancellation-flow

IanRamosC committed:
Jetpack Cancellation Flow: Fix unit of time translation contexts

IanRamosC committed:
Jetpack Cancellation Flow: Fix unit of time translation contexts

IanRamosC committed:
Jetpack Cancellation Flow: Fix usage of isJetpackSearch

IanRamosC committed:
Jetpack Cancellation Flow: Fix usage of isJetpackSearch

IanRamosC committed:
Merge branch 'add/jetpack-cancellation-flow-benefits-step' of github.com:Automattic/wp-calypso into update/improve-jetpack-cancellation-flow

IanRamosC committed:
Marketing Survey: Add text input for option 'other'

IanRamosC committed:
Marketing Survey: Fix styles for mobile devices

IanRamosC committed:
Marketing Survey: Add submitSurvey call

IanRamosC committed:
Marketing Survey: Add survey for improved Jetpack cancellation flow

This component adds the survey that will be used as a step of the new Jetpack cancellation flow via Calypso.

jboland88 committed:
Remove erroneous comment

jboland88 committed:
Remove erroneous comment

cbauerman commented on client/blocks/jetpack-benefits/feature-checks.ts:
```diff
@@ -0,0 +1,79 @@
+		// check plans for Jetpack search features
+		( isJetpackPlanSlug( productSlug ) &&
+			( planHasFeature( productSlug, FEATURE_JETPACK_SEARCH ) ||
+				planHasFeature( productSlug, FEATURE_JETPACK_SEARCH_MONTHLY ) ||
```
seeing this repeated pattern makes me thing it would be safe to add a `planHasAtLeastOneFeature()` function to `packages/calypso-products/src/main.js`

jboland88 replied:
Great suggestion, thanks! Opening a separate PR for this: https://github.com/Automattic/wp-calypso/pull/55581

cbauerman commented on client/blocks/jetpack-benefits/index.tsx:
```diff
@@ -0,0 +1,110 @@
+		return getRewindState( state, siteId );
+	} );
+
+	const scanState = useSelector( ( state ) => {
```
short arrow function syntax is preferred in Calypso

cbauerman commented on client/blocks/jetpack-benefits/site-backups.tsx:
```diff
@@ -0,0 +1,182 @@
+	} );
+
+	const getBackupTimeAgo = ( backup: { last_updated: string } ) => {
+		return moment.utc( backup.last_updated ).fromNow();
```
Calypso has a moment provider, so accessing moment via `const moment = useLocalizedMoment();` is preferred.

cbauerman commented on config/production.json:
```diff
@@ -64,6 +64,7 @@
 		"jetpack/magic-link-signup": true,
 		"jetpack/only-realtime-products": false,
 		"jetpack/personal-plan": false,
+		"jetpack/product-cancellation-flow": false,
```
👍

cbauerman reviewed (commented):
I left a couple comments on this overall PR, but I think it looks really good 👍

jboland88 committed:
Added a wrapper inside the cancellation dialog for more consistent styling

jboland88 committed:
Improve styling for cancellation flow by adding common classes

jboland88 committed:
Switch to using a formatted header on the benefits step

jboland88 committed:
Switch to using a formatted header on the benefits step

jboland88 committed:
Merge branch 'trunk' into update/improve-jetpack-cancellation-flow

jboland88 committed:
Merge branch 'add/survey-for-improved-jetpack-cancellation-flow' into update/improve-jetpack-cancellation-flow

jboland88 committed:
Merge branch 'add/jetpack-cancellation-flow-benefits-step' into update/improve-jetpack-cancellation-flow

IanRamosC committed:
Marketing Survey: Move the the selectedAnswerId to a prop

IanRamosC committed:
Merge branch 'add/survey-for-improved-jetpack-cancellation-flow' into update/improve-jetpack-cancellation-flow

jboland88 committed:
Refactor selector hooks to use short arrow syntax, switch to using the moment provider hook over moment include

jboland88 committed:
Refactor selector hooks to use short arrow syntax, switch to using the moment provider hook over moment include

jboland88 committed:
Merge branch 'add/jetpack-benefits-components' into update/improve-jetpack-cancellation-flow

IanRamosC committed:
Jetpack Cancellation Flow: Fix import/order in benefits step

IanRamosC committed:
Merge branch 'add/jetpack-cancellation-flow-benefits-step' into update/improve-jetpack-cancellation-flow

jboland88 committed:
Fix a few small typos

jboland88 committed:
Fix small typo

jboland88 committed:
Merge branch 'add/jetpack-cancellation-flow-benefits-step' of github.com:Automattic/wp-calypso into add/jetpack-cancellation-flow-benefits-step

jboland88 committed:
Merge branch 'add/jetpack-cancellation-flow-benefits-step' into update/improve-jetpack-cancellation-flow

jboland88 committed:
Merge branch 'add/jetpack-general-benefits-component' into update/improve-jetpack-cancellation-flow

a8ci18n commented on client/blocks/jetpack-benefits/general-benefits.tsx:
```diff
@@ -0,0 +1,159 @@
+	if ( hasBusinessPlan || hasSecurityRealTimePlan || hasCompletePlan ) {
+		benefits.push(
+			<React.Fragment>
+				{ translate( 'Unlimited {{strong}}high-speed video hosting{{/strong}}.', {
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **1** times:
`translate( '{{strong}}Unlimited{{/strong}} Video hosting' )` *ES Score: 6*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/improve-jetpack-cancellation-flow)

keoshi commented:
Took this for a quick spin and it all looks and feels really nice.

Only have a couple of comments:

### Flow

- In the screen we're asking the reason for cancelling, we should aim at not having any buttons at the bottom. As soon as I saw this I got decision paralysis, and we want the survey to be filled with no stress.

![image](https://user-images.githubusercontent.com/390760/130249285-542be934-b50f-4a1e-99ea-ed7d8a0c472a.png)

- Clicking any item on this screen should jump them forward in the process, instead of having to select one (first click/tap) and confirm their selection (second click/tap). There's an obvious exception which is the last one, where we can enter some details and then click the icon to submit.
- I did not see a cancellation offer after the step above — how can I test that step?


### Buttons

- The buttons (copy and styling) don't match the latest designs on pbNhbs-10d-p2. I understand we probably shouldn't have the black buttons, but there should be a primary one (“I'll keep it”) and a secondary one (“Next step”).
- Also note we're using sentence case for the all the buttons in Calypso rather than title case.

Live branch | Designs
--|--
![image](https://user-images.githubusercontent.com/390760/130248791-4180ec45-dd4c-4fde-9e4e-a5c5a8bc882f.png) | ![image](https://user-images.githubusercontent.com/390760/130248892-aa64527e-1172-4842-8cb0-a65ac6a6b1d6.png)


### Benefits

- Some of the benefits became misaligned when viewed in a smaller device, but since this is still in progress I don't know how relevant my comment is. :)

Thanks for doing this, @jboland88 ! Feel free to ping me again if you need another review in a few days.

jboland88 committed:
Merge branch 'trunk' into update/improve-jetpack-cancellation-flow

IanRamosC committed:
Merge branch 'trunk' into update/improve-jetpack-cancellation-flow

IanRamosC committed:
Address latest feedback

IanRamosC committed:
Merge branch 'add/jetpack-benefits-components' into update/improve-jetpack-cancellation-flow

IanRamosC committed:
Merge branch 'trunk' into update/improve-jetpack-cancellation-flow

IanRamosC committed:
Dialog: Fix buttons relevance classes

IanRamosC committed:
Enable Jetpack cancellation flow in staging

a8ci18n commented on client/components/marketing-survey/cancel-jetpack-form/index.tsx:
```diff
@@ -0,0 +1,283 @@
+		const prev = {
+			action: 'prev',
+			disabled,
+			label: translate( 'Previous step' ),
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **45** times:
`translate( 'Previous Step' )` *ES Score: 11*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/improve-jetpack-cancellation-flow)

keoshi reviewed (commented):
Hi @IanRamosC — I don't know if I'm seeing the latest changes on the live URL because it seems like the second step of the flow still has some issues.

The main one is having three buttons at the bottom:

![image](https://user-images.githubusercontent.com/390760/131502074-47a6f528-954a-4398-bc20-d3f9b151d3fb.png)

There's no useful use case where a customer needs to go back, so I'd personally remove the “Previous step” button.

The second issue is that we still require them to click/tap on “Cancel my plan” to confirm the reason for cancelling:

![image](https://user-images.githubusercontent.com/390760/131502332-d917cc59-6d0f-4222-ba37-e97f99437dfd.png)

This correlation between the two things is not apparent to me so I'd suggest doing the following:

* Disabling the button until a selection is made at the top.
* Getting rid of the chevrons (`>`) in the list.

IanRamosC committed:
Merge branch 'trunk' into update/improve-jetpack-cancellation-flow

IanRamosC committed:
Jetpack Cancellation flow: Address feedback

IanRamosC committed:
Jetpack Cancellation flow: Fix survey response not required to cancel

keoshi reviewed (approved):
This is looking great, @IanRamosC !

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/6528777

Thank you @jboland88 for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/6528777).
